benchmarks/src/main/java/org/opensearch/benchmark/index/mapper/FlatObjectMappingBenchmark.java

```diff
@@ -0,0 +1,363 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.benchmark.index.mapper;
+
+import org.apache.http.HttpHost;
+import org.json.JSONObject;
+import org.openjdk.jmh.annotations.Benchmark;
+import org.openjdk.jmh.annotations.BenchmarkMode;
+import org.openjdk.jmh.annotations.Fork;
+import org.openjdk.jmh.annotations.Level;
+import org.openjdk.jmh.annotations.Measurement;
+import org.openjdk.jmh.annotations.Mode;
+import org.openjdk.jmh.annotations.OutputTimeUnit;
+import org.openjdk.jmh.annotations.Scope;
+import org.openjdk.jmh.annotations.Setup;
+import org.openjdk.jmh.annotations.State;
+import org.openjdk.jmh.annotations.TearDown;
+import org.openjdk.jmh.annotations.Warmup;
+import org.opensearch.action.admin.indices.delete.DeleteIndexRequest;
+import org.opensearch.action.admin.indices.refresh.RefreshRequest;
+import org.opensearch.action.admin.indices.refresh.RefreshResponse;
+import org.opensearch.action.index.IndexRequest;
+import org.opensearch.action.index.IndexResponse;
+import org.opensearch.action.search.SearchRequest;
+import org.opensearch.action.search.SearchResponse;
+import org.opensearch.action.support.master.AcknowledgedResponse;
+import org.opensearch.benchmark.index.mapper.FlatObjectMappingBenchmark.MyState;
+import org.opensearch.client.RequestOptions;
+import org.opensearch.client.RestClient;
+import org.opensearch.client.RestHighLevelClient;
+import org.opensearch.client.indices.CreateIndexRequest;
+import org.opensearch.client.indices.CreateIndexResponse;
+import org.opensearch.common.xcontent.XContentFactory;
+import org.opensearch.common.xcontent.XContentType;
+import org.opensearch.index.query.QueryBuilders;
+import org.opensearch.search.SearchHits;
+import org.opensearch.search.builder.SearchSourceBuilder;
+
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.Random;
+import java.util.concurrent.TimeUnit;
+
+@State(Scope.Thread)
+@Fork(1)
+@Warmup(iterations = 1, time = 1, timeUnit = TimeUnit.SECONDS)
+@Measurement(iterations = 1, time = 1, timeUnit = TimeUnit.SECONDS)
+
+public class FlatObjectMappingBenchmark {
+
+    @State(Scope.Thread)
+    public static class MyState {
+        private RestHighLevelClient client;
```

Comment:
I don't actually know the right answer here, but the benchmarks here are generally "micro benchmarks" that target pretty low-level functionality and aren't making remote calls. Would the more "macro" use case you're targeting here be a better fit for [opensearch-benchmark](https://github.com/opensearch-project/opensearch-benchmark)?

Reply:
+1 to that, we have dedicated project for high level benchmarks (those are not microbenchmarks)

Reply:
Agree that this benchmarking class is not micro, which is set to run benchmarking performance between dynamic mapping and flat_object mapping. As we recorded the performance outcomes, it can be moved out from the micro benchmark folder. We will discuss offline and see if that's necessary to push to [opensearch-benchmark](https://github.com/opensearch-project/opensearch-benchmark) after this feature get released.

Reply:
Moved out the microbenchmark class.